mdl/src/main/cft/mdlCreateS3Buckets.yml

```diff
@@ -95,6 +95,84 @@ Resources:
               - - 'arn:aws:s3:::'
                 - !If [EnableSSLAndAuth, !Ref CloudfrontS3Bucket, !Ref S3BucketForStaticPages]
                 - /*
+  LambdaExecutionRole:
+    Type: 'AWS::IAM::Role'
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - lambda.amazonaws.com
+            Action:
+              - sts:AssumeRole
+      Path: /
+      Policies:
+        - PolicyName: root
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action:
+                  - logs:CreateLogGroup
+                  - logs:CreateLogStream
+                  - logs:PutLogEvents
+                Resource: arn:aws:logs:*:*:*
+              - Effect: Allow
+                Action:
+                  - s3:Get*
+                  - s3:List*
+                  - s3:DeleteObject
+                Resource: '*'
+  EmptyS3LambdaFunction:
+    Type: AWS::Lambda::Function
+    Properties:
+      Handler: 'index.handler'
+      Code:
+        ZipFile: |
+          import cfnresponse
+          import boto3
+
+          def handler(event, context):
+              bucket_list = event['ResourceProperties']['BucketList']
+              physical_id = "EmptyS3BucketLambdaPhysicalID"
+              response_data = {}
+              if event['RequestType'] == 'Delete':
+                  try:
+                      for bucket_name in bucket_list:
+                          print('Emptying s3 bucket:{}'.format(bucket_name))
+                          empty_s3_bucket(bucket_name)
+                  except Exception as e:
+                      response_data['Message'] = 'Unexpected error. Exception: {}'.format(e)
+                      cfnresponse.send(event, context, cfnresponse.FAILED, response_data, physical_id)
+                  response_data['Message'] = 'Succssfully Empty S3 bucket list: {}'.format(bucket_list)
+                  cfnresponse.send(event, context, cfnresponse.SUCCESS, response_data, physical_id)
+              else:
+                  response_data['Message'] = 'Nothing to do.'
+                  cfnresponse.send(event, context, cfnresponse.SUCCESS, response_data, physical_id)
+
+          def empty_s3_bucket(bucket_name):
+              client = boto3.client( 's3')
+              paginator = client.get_paginator( 'list_objects' )
```

Comment:
This could be done in a simpler way:

```
bucket = s3.Bucket(bucket_name)
for key in bucket.objects.all():
    key.delete()
```

Reply:
yelp, this is simpler and quicker, can't believe I use this even in my personal util too.  but I copied the old empty s3 codes, and then it doesn't work for more than 1k items, so I added pagination deletion based on that...